### PR TITLE
Add log level setting

### LINKED ISSUE #163: Logging issue
I've followed the discussion in https://github.com/aws/aws-xray-sdk-node/issues/2 and tried to set my own logger at a level of error to stop receiving the following logs without any success.

`
AWS_XRAY_DAEMON_ADDRESS is set. Configured daemon address to 169.254.79.2:2000.
AWS_XRAY_CONTEXT_MISSING is set. Configured context missing strategy to LOG_ERROR.
Subsegment streaming threshold set to: 0
`

We setup the X-Ray as below
`
const AWSSDK = require('aws-sdk');
const AWSXRay = require('aws-xray-sdk-core');
AWSXRay.setLogger(loggerForXRAY);
AWSXRay.captureAWS(AWSSDK)
`

Is this something I've approached in the wrong way or is there some other issue?

### PR DESCRIPTION
**Pull request by grahamhar**
Attempt to fix #163 
In the lambda execution environment having xray info logs adds noise and
potentially cost, this add the ability to set the log level via
environment variable allowing these to be suppressed.

### CODE REVIEW

File: packages/core/test/unit/logger.test.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    logger = require(path);
+  }
+
+  describe('defaults', function () {
+    afterEach(function () {
+      delete process.env.AWS_XRAY_DEBUG_MODE;
+      reloadLogger();
+
+    });
+    it('Should have a default logger with no transport and level set to info', function () {
```
Comment (willarmiros) [inline packages/core/test/unit/logger.test.js]: This test fails if the machine from which it is run has the 'AWS_XRAY_LOG_LEVEL' environment variable set already. I recommend setting it equal to the empty string then reloading the logger as you do in other tests before the assertions.

Comment (grahamhar) [reply packages/core/test/unit/logger.test.js]: Good spot, I've addressed this I think.

Also I think the same scenario could have happened in the final test so explicitly unset AWS_XRAY_DEBUG_MODE there.

Comment (willarmiros) [reply packages/core/test/unit/logger.test.js]: Tests pass in all cases now. Thanks for the fix.

**Comment by grahamhar** [comment]
@willarmiros do you have any idea when this will be available in published version?

**Comment by willarmiros** [comment]
Unfortunately I cannot provide a timeline on future releases.

**Pull request merged**